src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/helper/ErrorResponseHelper.kt

```diff
@@ -4,7 +4,7 @@ import org.assertj.core.api.Assertions.assertThat
 import org.springframework.http.HttpStatus
 import org.springframework.test.web.reactive.server.WebTestClient
 
-fun <T : WebTestClient.ResponseSpec> T.expectErrorResponse(status: HttpStatus, vararg messages: String): T {
+fun <T : WebTestClient.ResponseSpec> T.expectErrorResponse(status: HttpStatus, vararg messages: String, errorCode: Int? = null): T {
```

Comment:
could `errorCode` have `ErrorCode?` type? (but still check the value/number?)

Reply:
this was actually deliberate cos those numbers in the enum are too easy to change. if someone in future adds a new constant to the error codes and decides to renumber things, they'll get a lot of tests failing. i wanted it to be clear that you break something if you change the numbers.

Reply:
i agree that it's not as elegant and makes the tests more tedious to read and write, but i think they're a bit safer that way… what do you think now?

Reply:
🤔 I suppose there is always a risk of something changing but with these enums they should be considered as part of the API contract almost.

Maybe there should be tests to check the code and the type/value match and then use the enum everywhere else 🤷 but maybe it feels like faff?